### PR TITLE
Allow pre-release (EKS) kubernetes versions

### LINKED ISSUE #69: Helm Chart kubeVersion refuses to install on EKS
The current version of EKS (`v1.30.1-eks-1de2ab1` at the time of writing) does not pass the semver test because EKS (incorrectly) uses a pre-release version (see https://github.com/helm/helm/issues/9371).

This is preventing the Helm chart from installing on EKS.

### PR DESCRIPTION
**Pull request by alanraison**
As described in https://github.com/helm/helm/issues/9371, adding `-0` to the SemVer string allows pre-release versions to be included in the permitted versions. This is necessary for installing the chart on EKS.

Closes #69

### CODE REVIEW

**Comment by ordovicia** [comment]
Hmm, my understanding is:

- `v1.30.1-eks-1de2ab1` is a pre-release version. EKS should use `v1.30.1+eks.1de2ab1` instead to stick to SemVer spec. Amazon and other cloud providers does not follow SemVer habitually.
- Helm and other tools, by default, ignores pre-release versions when determining if a version is in a range. It is correct behavior in the light of SemVer semantics.
- By adding `-0` to SemVer range, we can make Helm count in pre-release versions.

File: charts/gcp-workload-identity-federation-webhook/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: gcp-workload-identity-federation-webhook
 description: A Helm chart for gcp-workload-identity-federation-webhook
-kubeVersion: "^1.21.0"
+kubeVersion: "^1.21.0-0"
```
Comment (ordovicia) [inline charts/gcp-workload-identity-federation-webhook/Chart.yaml:6]: Could you please add some comments and a link to helm/helm#9371 to describe the situation and why we need this workaround?

Comment (alanraison) [reply charts/gcp-workload-identity-federation-webhook/Chart.yaml:6]: Sorry, where do you mean for me to add comments? In the Chart or in the PR?

**Comment by alanraison** [comment]
> Hmm, my understanding is:
> 
> * `v1.30.1-eks-1de2ab1` is a pre-release version. EKS should use `v1.30.1+eks.1de2ab1` instead to stick to SemVer spec. Amazon and other cloud providers does not follow SemVer habitually.
> * Helm and other tools, by default, ignores pre-release versions when determining if a version is in a range. It is correct behavior in the light of SemVer semantics.
> * By adding `-0` to SemVer range, we can make Helm count in pre-release versions.

Yes this is correct. By using the tool [vert](https://github.com/masterminds/vert), you can test the differences;

```
> vert ^1.20.0 v1.30.2-eks-db838b0
> echo $?
1
> vert ^1.20.0-0 v1.30.2-eks-db838b0
1.30.2-eks-db838b0
> echo $?
0
```

**Comment by superbrothers** [comment]
It may be possible to remove the `kubeVersion` itself. If we install Chart and it works fine, we can use it.

File: charts/gcp-workload-identity-federation-webhook/Chart.yaml (hunk 1 above)
Comment (ordovicia) [reply charts/gcp-workload-identity-federation-webhook/Chart.yaml:6]: I mean adding comments to the Chart.
Chart users would be able to understand why `kubeVersion` includes `-0` without navigating the Git history to this PR.

**Review by ordovicia — APPROVED** [review]

**Pull request merged**